libcxxabi fails to compile for ARM/FreeBSD and ARM/OpenBSD in 22.x

```
/home/alexrp/Sources/ziglang/zig/lib/libcxxabi/src/cxa_personality.cpp:1135:11: error: use of undeclared identifier
      '__gnu_unwind_frame'
 1135 |   switch (__gnu_unwind_frame(unwind_exception, context)) {
      |           ^~~~~~~~~~~~~~~~~~
1 error generated.
```

Notably, 3c100d5d548d511d09746dd5a2fee29bdd00805c removed the `__gnu_unwind_frame` declaration in that file. cc @philnik777

--- a/libcxxabi/src/cxa_personality.cpp
+++ b/libcxxabi/src/cxa_personality.cpp
@@ -1116,6 +1116,8 @@ __gxx_personality_seh0(PEXCEPTION_RECORD ms_exc, void *this_frame,
 
 #else
 
+extern "C" _Unwind_Reason_Code __gnu_unwind_frame(_Unwind_Exception*, _Unwind_Context*);
+
 // Helper function to unwind one frame.
 // ARM EHABI 7.3 and 7.4: If the personality function returns _URC_CONTINUE_UNWIND, the
 // personality routine should update the virtual register set (VRS) according to the

